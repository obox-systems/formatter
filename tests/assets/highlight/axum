use axum::{handler::get, Router};
use std::net::SocketAddr;

async fn hello_world() -> &'static str {
    "Hello, World!"
}

#[tokio::main]
async fn main() {
    // Create a new Axum router
    let app = Router::new()
        .route("/", get(hello_world));

    // Set the server address
    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));

    // Start the server
    axum::Server::bind(&addr)
        .serve(app.into_make_service())
        .await
        .unwrap();
}
