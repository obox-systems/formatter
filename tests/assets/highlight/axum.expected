[42muse[0m[47m [0m[42maxum[0m[42m:[0m[42m:[0m[100m{[0m[42mhandler[0m[42m:[0m[42m:[0m[42mget[0m[40m,[0m[47m [0m[42mRouter[0m[100m}[0m[40m;[0m[101m
[0m[42muse[0m[47m [0m[42mstd[0m[42m:[0m[42m:[0m[42mnet[0m[42m:[0m[42m:[0m[42mSocketAddr[0m[40m;[0m[101m

[0m[42masync[0m[47m [0m[42mfn[0m[47m [0m[42mhello_world[0m[100m([0m[100m)[0m[47m [0m[42m->[0m[47m [0m[42m&[0m[42m'static[0m[47m [0m[42mstr[0m[47m [0m[100m{[0m[101m
[0m[47m    [0m[105m"Hello, World!"[0m[101m
[0m[100m}[0m[101m

[0m[40m#[0m[100m[[0m[42mtokio[0m[42m:[0m[42m:[0m[42mmain[0m[100m][0m[101m
[0m[42masync[0m[47m [0m[42mfn[0m[47m [0m[42mmain[0m[100m([0m[100m)[0m[47m [0m[100m{[0m[101m
[0m[47m    [0m[46m// Create a new Axum router[0m[101m
[0m[47m    [0m[42mlet[0m[47m [0m[42mapp[0m[47m [0m[42m=[0m[47m [0m[42mRouter[0m[42m:[0m[42m:[0m[42mnew[0m[100m([0m[100m)[0m[101m
[0m[47m        [0m[40m.[0m[45mr[0m[42moute[0m[100m([0m[105m"/"[0m[40m,[0m[47m [0m[42mget[0m[100m([0m[42mhello_world[0m[100m)[0m[100m)[0m[40m;[0m[101m

[0m[47m    [0m[46m// Set the server address[0m[101m
[0m[47m    [0m[42mlet[0m[47m [0m[42maddr[0m[47m [0m[42m=[0m[47m [0m[42mSocketAddr[0m[42m:[0m[42m:[0m[42mfrom[0m[100m([0m[100m([0m[100m[[0m[40m1[0m[40m2[0m[40m7[0m[40m,[0m[47m [0m[40m0[0m[40m,[0m[47m [0m[40m0[0m[40m,[0m[47m [0m[40m1[0m[100m][0m[40m,[0m[47m [0m[40m3[0m[40m0[0m[40m0[0m[40m0[0m[100m)[0m[100m)[0m[40m;[0m[101m

[0m[47m    [0m[46m// Start the server[0m[101m
[0m[47m    [0m[42maxum[0m[42m:[0m[42m:[0m[42mServer[0m[42m:[0m[42m:[0m[42mbind[0m[100m([0m[42m&[0m[42maddr[0m[100m)[0m[101m
[0m[47m        [0m[40m.[0m[42mserve[0m[100m([0m[42mapp[0m[40m.[0m[42minto_make_service[0m[100m([0m[100m)[0m[100m)[0m[101m
[0m[47m        [0m[40m.[0m[42mawait[0m[101m
[0m[47m        [0m[40m.[0m[42munwrap[0m[100m([0m[100m)[0m[40m;[0m[101m
[0m[100m}[0m[101m
[0m